cookbooks/aws-parallelcluster-environment/recipes/init/backup_internal_use_shared_data.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+#
+# Copyright:: 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+return if on_docker?
+
+case node['cluster']['node_type']
+when 'HeadNode'
+  Chef::Log.info("Internal Shared Storage #{node['cluster']['internal_shared_dirs']}")
+  # For each, backup the data to a temp location
+  node['cluster']['internal_shared_dirs'].each do |dir|
+    bash "Backup #{dir}" do
+      user 'root'
+      group 'root'
+      code <<-EOH
+        mkdir -p /tmp#{dir}
+        rsync -a #{dir}/ /tmp#{dir}
+      EOH
+    end
+  end
+
+when 'ComputeFleet', 'LoginNode'
+  Chef::Log.info("Backup only from the HeadNode")
+else
+  raise "node_type must be HeadNode, ComputeFleet, or LoginNode"
+end
```

Comment:
I suggest one of the following two options in replacement of this section:
1. omit this section as a whole because we may consider redundant to log the skip of an action
2. have only `when HeadNode { do action } else { Log(Skipping action) }`; the advantage is that if we add other node types we do not need to refactor the statement.

Personally I'm more in favour of option 1 because it removes noise.

Reply:
I'm actually more in favor of removing the else statement and only having the if.  Do we really need to guard against missing node types?

Reply:
I should also note that this precedent was sent a couple months ago https://github.com/aws/aws-parallelcluster-cookbook/blob/develop/cookbooks/aws-parallelcluster-environment/recipes/config/mount_shared.rb

Reply:
Yup, it’s part of the latest version of the pr